tsdb/index/postings.go

```diff
@@ -415,18 +415,44 @@ func (p *MemPostings) PostingsForLabelMatching(ctx context.Context, name string,
 		}
 	}
 
-	var its []Postings
```

Comment:
There's a comment above saying:

```
// Benchmarking shows that first copying the values into a slice and then matching over that is
// faster than matching over the map keys directly, at least on AMD64.
```

That copying is now essential to avoid a race, given we're releasing the lock. We should update the comment accordingly.

Reply:
Updated in https://github.com/prometheus/prometheus/pull/14286/commits/00ae61acbbbaa7d66fa5f30781cac65cebd52f48